### PR TITLE
Focus input when opening and remove autocomplete

### PR DESCRIPTION
**Pull request by anthonydillon**
Fixes: https://github.com/juju/juju-gui/issues/1352

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4500/
Test FAILed.

**Comment by anthonydillon** [comment]
## QA
- Add a service and go into Units in the inspector
- Click Scale service
- The units input should auto focus and should only accept numbers
- Now go to machine view
- Click on the orange + by New units to scale
- The unit inputs here should only accept number but look the same as text inputs

File: jujugui/static/gui/src/app/assets/css/base.scss [hunk 1]
```diff
@@ -164,6 +165,14 @@ input[type="password"] {
     border-radius: $border-radius;
 }
 
+input[type=number]::-webkit-inner-spin-button,
+input[type=number]::-webkit-outer-spin-button {
```
Comment (hatched) [inline jujugui/static/gui/src/app/assets/css/base.scss]: Does this mean we actually can style them? Or only in certain browsers? I could see them be there if we could get some styling that matched the UI

**Comment by anthonydillon** [comment]
Fixes: https://github.com/juju/juju-gui/issues/1352

File: jujugui/static/gui/src/app/components/scale-service/scale-service.js [hunk 2]
```diff
@@ -130,10 +130,16 @@ YUI.add('scale-service', function() {
...
-              type="text"
+              type="number"
+              autoComplete="off"
               name="num-units"
               onChange={this._updateState}
-              ref="numUnitsInput"/>
+              ref={function(input) {
+                if (input != null) {
+                  input.focus();
+                  return "numUnitsInput";
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/scale-service/scale-service.js]: This is quite interesting, for this to work that means that the ref is calculated after the element is in the DOM. Good to know.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4501/
Test FAILed.

File: jujugui/static/gui/src/app/assets/css/base.scss (hunk 1 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/base.scss]: Only modern browser support number functionality so we should be able to style them all but I will browser test to see if there will be arrows.

**Comment by hatched** [comment]
:+1: :ok: with trivial issue below

QA Issue:
An interesting side effect of the number input is that it will create negative numbers. Can you lock the lower bound please.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4503/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4504/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4505/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4506/
Test PASSed.

**Comment by kadams54** [comment]
:+1:

**Comment by kadams54** [comment]
Thanks @anthonydillon! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**